File: SpringBoot-Simple-JPA/src/main/java/com/example/springbootsimplejpa/repository/OrderRepository.java
package com.example.springbootsimplejpa.repository;

import com.example.springbootsimplejpa.domain.OrderModel;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface OrderRepository extends JpaRepository<OrderModel, String> {
    OrderModel findByOrderCode(String orderCode);
}

File: SpringKotlin-Simple-Example/src/main/kotlin/com/example/springkotlinsimpleexample/repository/OrderRepository.kt
package com.example.springkotlinsimpleexample.repository

import com.example.springkotlinsimpleexample.domain.OrderModel
import org.springframework.data.repository.CrudRepository

interface OrderRepository : CrudRepository<OrderModel, String> {

    fun findAllByOrderUserName(orderCode: String): List<OrderModel>

    fun findByOrderCode(orderCode: String): OrderModel?
}
